src/main/java/net/fabricmc/loader/impl/transformer/ClassStripper.java

```diff
@@ -58,13 +62,41 @@ public void visit(int version, int access, String name, String signature, String
 
 	@Override
 	public FieldVisitor visitField(int access, String name, String descriptor, String signature, Object value) {
-		if (stripFields.contains(name + descriptor)) return null;
-		return super.visitField(access, name, descriptor, signature, value);
+		if (stripFields.contains(name + descriptor)) {
+			return null;
+		} else {
+			return super.visitField(access, name, descriptor, signature, value);
+		}
 	}
 
 	@Override
 	public MethodVisitor visitMethod(int access, String name, String descriptor, String signature, String[] exceptions) {
-		if (stripMethods.contains(name + descriptor)) return null;
-		return super.visitMethod(access, name, descriptor, signature, exceptions);
+		if (stripMethods.contains(name + descriptor)) {
+			return null;
+		}
+
+		MethodVisitor ret = super.visitMethod(access, name, descriptor, signature, exceptions);
+
+		// strip field writes from static init and constructor
+		if (!stripFields.isEmpty() && (name.equals("<clinit>") || name.equals("<init>"))) {
```

Comment:
This will strip writes to other instances, will strip non-ctor writes to instance fields, but only if they're in <clinit>, and vice versa. Seems confusing and is mostly avoidable.

Reply:
It seems odd for a non-static write to be stripped from `<clinit>` but not from any other non-ctor methods.

Reply:
I think stripping PUTSTATIC from clinit and PUTFIELD from init would be very little effort and make far more sense.